dkg/pedersen/handler.go

```diff
@@ -530,3 +607,299 @@ func (h *Handler) announceDkgPublicKey(out mino.Sender, from mino.Address) error
 
 	return nil
 }
+
+// handles the resharing request - acts differently for the new and old and common nodes
+func (h *Handler) handleReshare(out mino.Sender,
+	from mino.Address, msg types.ResharingRequest, done chan struct{}) error {
+	// if the node is completely new
+	if h.startRes.distrKey == nil {
+		if h.startRes.Done() {
+			dela.Logger.Warn().Msgf(
+				"%v ignored reshare request from %v as it has"+
+					" started before\n", h.me, from)
+			return xerrors.Errorf("dkg is started")
+		}
+
+	} else { // if the node is in the old committee or is a common node
+		if !h.startRes.Done() {
```

Comment:
this translates to `h.startRes.participants == nil` , given we already know that `distrKey != nil` at this point.
I find the state management a bit confusing here.

It might be helpful to make it explicit in the state that we're one of:
Uninitialized / Starting / Ready / Joining [or] Resharing / Ready [or] Finished
with Joining being for the case where we're a new node in a reshare,
and Finished being for the case where we're not part of a committee anymore.